examples/out_lib/out_lib.c

```diff
@@ -30,13 +32,47 @@ int my_stdout_json(void *record, size_t size, void *data)
     return 0;
 }
 
-int my_stdout_msgpack(void* data, size_t size)
+int my_stdout_msgpack(void *record, size_t size, void *data)
 {
+    msgpack_object obj = *(msgpack_object*) record;
+
     printf("[%s]",__FUNCTION__);
-    msgpack_object_print(stdout, *(msgpack_object*)data);
-    printf("\n");
 
-    flb_lib_free(data);
+    msgpack_unpacked result;
+    size_t off = 0, cnt = 0;
+    struct flb_time tmp_time;
+    msgpack_object *x;
+
+    msgpack_unpacked_init(&result);
+
+    while (msgpack_unpack_next(&result, record, size, &off) == MSGPACK_UNPACK_SUCCESS) {
+        printf("[%zd]: [", cnt++);
+        flb_time_pop_from_msgpack(&tmp_time, &result, &x);
+        printf("%"PRIu32".%09lu, ", (uint32_t)tmp_time.tm.tv_sec, tmp_time.tm.tv_nsec);
+
+        if(x->type == MSGPACK_OBJECT_MAP){
+            fprintf(stdout, "{");
+            if(x->via.map.size != 0) {
+                msgpack_object_kv* p = x->via.map.ptr;
+                msgpack_object_kv* const pend = x->via.map.ptr + x->via.map.size;
+                do{
+                    if (p->key.type != MSGPACK_OBJECT_STR || p->val.type != MSGPACK_OBJECT_STR){
+                        break;
+                    }
+                    msgpack_object_print(stdout, p->key);
+                    fprintf(stdout, "=>");
+                    msgpack_object_print(stdout, p->val);
+                    ++p;
+                } while(p < pend);
+            }
+            fprintf(stdout, "}");
+        }
```

Comment:
This lines can be replaced by a single line
```c
        msgpack_object_print(stdout, *x);
```

Reply:
True, but this way it demonstrates how to extract and use keys and values from the msgpack object.

Reply:
OK.